src/Model/Item.php

```diff
@@ -1508,28 +1516,54 @@ public static function storeForUserByUriId(int $uri_id, int $uid, array $fields
 		if (($uid != 0) && (($item['gravity'] == GRAVITY_PARENT) || $is_reshare) &&
 			DI::pConfig()->get($uid, 'system', 'accept_only_sharer') == self::COMPLETION_NONE &&
 			!in_array($item['post-reason'], [self::PR_FOLLOWER, self::PR_TAG, self::PR_TO, self::PR_CC])) {
-			Logger::info('Contact is not a follower, thread will not be stored', ['author' => $item['author-link'], 'uid' => $uid]);
+			Logger::info('Contact is not a follower, thread will not be stored', ['author' => $item['author-link'], 'uid' => $uid, 'uri-id' => $uri_id]);
 			return 0;
 		}
 
-		if (($uri_id != $item['thr-parent-id']) && (($item['gravity'] == GRAVITY_COMMENT) || $is_reshare) && !Post::exists(['uri-id' => $item['thr-parent-id'], 'uid' => $uid])) {
-			// Fetch the origin user for the post
-			$origin_uid = self::GetOriginUidForUriId($item['thr-parent-id'], $uid);
-			if (is_null($origin_uid)) {
-				Logger::info('Origin item was not found', ['uid' => $uid, 'uri-id' => $item['thr-parent-id']]);
+		$causer = $item['causer-id'] ?: $item['author-id'];
+
+		if (($uri_id != $item['parent-uri-id']) && (($item['gravity'] == GRAVITY_COMMENT) || $is_reshare) && !Post::exists(['uri-id' => $item['parent-uri-id'], 'uid' => $uid])) {
+			if (!self::fetchParent($item['parent-uri-id'], $uid, $causer)) {
```

Comment:
Aren’t we going to run into PHP recursion limits once more?

Reply:
Less than before.